Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -701,11 +711,31 @@ private ResultsViewModel SelectedResults
                     // result from the one that was selected before going into the context menu to the first result.
                     // The code below correctly restores QueryText and puts the text caret at the end without
                     // running the query again when returning from the context menu.
-                    if (isReturningFromContextMenu)
+                    if (isReturningFromContextMenu && _lastSelectedResultItem != null)
                     {
-                        _queryText = _queryTextBeforeLeaveResults;
-                        OnPropertyChanged(nameof(QueryText));
-                        QueryTextCursorMovedToEnd = true;
+                        try
+                        {
+                            _blockQueryExecution = true;
+                            // Set querytext without running the query for keep index. if not use this, index will be 0.
+                            _queryText = _queryTextBeforeLeaveResults;
```

Comment:
why not just change QueryText

Reply:
I noted this in the PR description:

If you simply update the `queryText`, the index resets to 0 when the results are refreshed. The flow goes like this:

1. Remember the previously selected item  
2. Open the context menu (which clears the query box text)  
3. Close the context menu (the previous query text is restored)

Even if the selected index is correctly remembered and restored at step 3, it becomes pointless because restoring the `queryText` causes the index to reset to 0 again.